src/main/kotlin/graphics/scenery/controls/SwingBridgeFrame.kt

```diff
@@ -0,0 +1,176 @@
+package graphics.scenery.controls
+
+
+import graphics.scenery.utils.Image
+import graphics.scenery.utils.SystemHelpers.Companion.logger
+import java.awt.Component
+import java.awt.event.*
+import java.awt.image.BufferedImage
+import javax.swing.JFrame
+import javax.swing.SwingUtilities
+
+/**
+ * @author Jan Tiemann
+ * @author Konrad Michel
+ *
+ * Bridge to link the Swing window events with scenery key inputs in combination with SwingUiNode
+ */
+open class SwingBridgeFrame(title: String) : JFrame(title) {
+
+    val uiNode = SwingUiNode(this)
+    var finalImage : Image? = null
+    var dragged = false
+    init {
+        this.addKeyListener(object : KeyListener {
+            override fun keyTyped(e : KeyEvent?) {
+            }
+            override fun keyPressed(e : KeyEvent?) {
+                updateImage()
+            }
+            override fun keyReleased(e : KeyEvent?) {
+            }
+        })
+    }
+    private fun updateImage()
+    {
```

Comment:
The way this is implemented, you'll always allocate a new buffer for the image. This is probably not what you want.